src/traits.rs

```diff
@@ -70,9 +74,12 @@ pub trait ServiceProtocol {
     fn received(&mut self, _context: ProtocolContextMutRef, _data: bytes::Bytes) {}
     /// Called when the Service receives the notify task
     fn notify(&mut self, _context: &mut ProtocolContext, _token: u64) {}
-    /// Behave like `Stream::poll`, but nothing output
+    /// Behave like `Stream::poll_next`, but nothing output, just know pending or not,
+    /// if ready, it will continue poll immediately
     #[inline]
-    fn poll(self: Pin<&mut Self>, _cx: &mut Context, _context: &mut ProtocolContext) {}
+    fn poll(self: Pin<&mut Self>, _cx: &mut Context, _context: &mut ProtocolContext) -> Poll<()> {
+        Poll::Pending
```

Comment:
What the protocol author should do when returning "Pending"? Who is responsible to ensure the task is notified? Is it too generous to provide a default implementation which just returns Pending?

Reply:
maybe here should use a `Poll<Option<()>>` to let parents know it should or not poll